tests/test_serialization.py

```diff
@@ -0,0 +1,63 @@
+from datetime import datetime, timedelta
+
+import pandas as pd
+import pytest
+
+from pangeo_forge_recipes.patterns import ConcatDim, FilePattern
+from pangeo_forge_recipes.serialization import match_pattern_blockchain
+
+URL_FORMAT = (
+    "https://www.ncei.noaa.gov/data/sea-surface-temperature-optimum-interpolation/"
+    "v2.1/access/avhrr/{time:%Y%m}/oisst-avhrr-v02r01.{time:%Y%m%d}.nc"
+)
+
+
+def make_file_pattern(dates, nitems_per_file):
+    def make_url(time):
+        return URL_FORMAT.format(time=time)
+
+    time_concat_dim = ConcatDim("time", dates, nitems_per_file=nitems_per_file)
+    pattern = FilePattern(make_url, time_concat_dim)
+
+    return pattern
+
+
+@pytest.fixture
+def end_date():
+    return "2022-02-01"
+
+
+@pytest.fixture
+def old_pattern_sha256(end_date):
+    dates = pd.date_range("1981-09-01", end_date, freq="D")
```

Comment:
Could make this test much faster by using a much shorter date range.

Reply:
👍 https://github.com/pangeo-forge/pangeo-forge-recipes/pull/349/commits/e34afdae8105a406068cd8441a43a7a0456a6b49